### PR TITLE
fixes #37

### LINKED ISSUE #37: hms class dropped when combining
```R
> c(hms(1:2), hms(3))
#>Time differences in secs
#> [1] 1 2 3
```
I expected combination of `hms` to return another `hms` object.
If it is a confirmed bug, I am quite happy to patch it

### PR DESCRIPTION
**Pull request by qgeissmann**
We can now use `c(hms(1), hms(2))` and get a hms object back.
`c(hms(1), NA)` will not work because it is not implemented is base's `difftime`. So relevant unittest commented out for now. Should be another issue.

### CODE REVIEW

File: tests/testthat/test-combine.R [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+test_that("combination coerces to hms", {
+  expect_identical(c(hms(1),2), hms(1:2))
+  expect_identical(c(hms(1),"00:00:02"), hms(1:2))
+})
+
+
+# this does not work in base R: `c(as.difftime("20:00:00"), NA)`
+# so not sure whether we want it to work in hms...
+# test_that("composition with NA works", {
+#   expect_identical(c(hms(1),NA), hms(c(1, NA)))
```
Comment (krlmlr) [inline tests/testthat/test-combine.R]: I'm fine with just having `expect_error()` here.

File: tests/testthat/test-combine.R [hunk 2]
```diff
@@ -0,0 +1,20 @@
+context("combine")
+
+test_that("combination keeps class and order", {
+  expect_identical(c(hms(1),hms(2)), hms(1:2))
```
Comment (krlmlr) [inline tests/testthat/test-combine.R]: Can you please add space after the commas?

File: tests/testthat/test-combine.R [hunk 3]
```diff
@@ -0,0 +1,19 @@
+context("combine")
+
+test_that("combination keeps class and order", {
+  expect_identical(c(hms(1), hms(2)), hms(1:2))
+})
+
+test_that("combination coerces to hms", {
+  expect_identical(c(hms(1), 2), hms(1:2))
+  expect_identical(c(hms(1), "00:00:02"), hms(1:2))
```
Comment (krlmlr) [inline tests/testthat/test-combine.R]: I'd expect this to throw an error. Base R converts to `NA` and warns, but we can be stricter here (and relax later if necessary):

``` r
class(as.POSIXct("2017-12-12 12:34:56 CEST"))
#> [1] "POSIXct" "POSIXt"
c(Sys.time(), "2017-12-12 12:34:56 CEST")
#> Warning in as.POSIXlt.POSIXct(x, tz): NAs introduced by coercion
#> [1] "2017-08-03 09:32:19 CEST" NA
```

**Comment by krlmlr** [comment]
Are you still planning to expand on this?

**Comment by qgeissmann** [comment]
Yes, sorry, was quite busy in the last few weeks. I will finish this next week if it is ok.

**Comment by krlmlr** [comment]
No worries. We should keep #42 in mind, too.

File: tests/testthat/test-combine.R (hunk 3 above)
Comment (qgeissmann) [reply tests/testthat/test-combine.R]: The behaviour for `difftime` is different though:

```R
c(as.difftime("20:00:00", units="sec"), "00:00:10")
#> Time differences in secs
#> [1] 72000    10
```
I think we want to use `as.hms()` inside `c()`, but `as.hms()` will converts strings as well.
If we coerce to numeric, e.g.: 

```R
c.hms <- function(x, ...) {
  hms(c(as.numeric(x), as.numeric(...)))
}
```
Thoughts?
We risk having inconsistent behaviours:
```R
d <- as.difftime("20:00:00")
c(hms(1), d)
#> 00:00:01
#> 00:00:20
```

Comment (krlmlr) [reply tests/testthat/test-combine.R]: Thanks. I wasn't aware of the difftime behavior.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**